formats/iso1.go

```diff
@@ -0,0 +1,134 @@
+package formats
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"strings"
+	"text/tabwriter"
+)
+
+type ISO1 struct {
+	Filler string
+
+	debugWriter io.Writer
+}
+
+func NewISO1() *ISO1 {
+	return &ISO1{
+		Filler: "F", // default to ISO1's Filler
+	}
+}
+
+// Format returns iso type
+func (i *ISO1) format() string {
+	return "Format 1 (ISO-1)"
+}
+
+// Padding returns padding pattern
+func (i *ISO1) padding(pin string) (string, error) {
+
+	if len(pin) < 4 {
+		return "", fmt.Errorf("pin length must be between 4 and 12 digits")
+	}
+
+	length := 14 - len(pin)
+	if i.Filler != "" {
+		return strings.Repeat(i.Filler, length), nil
+	} else {
+		randomBytes := make([]byte, 7)
+		_, err := rand.Read(randomBytes)
+		if err != nil {
+			return "", fmt.Errorf("generating random bytes: %w", err)
+		}
+		return hex.EncodeToString(randomBytes)[:length], nil
+	}
+}
+
+// SetDebugWriter will set writer for getting output message of encoding and decoding logic
+func (i *ISO1) SetDebugWriter(writer io.Writer) {
+	i.debugWriter = tabwriter.NewWriter(writer, 0, 0, 2, ' ', 0)
+}
+
+// Encode returns the ISO1 PIN block for the given PIN
+//
+//	The `ISO-1` PIN block format is equivalent to an `ECI-4` PIN block format
+//	and is recommended for usage where no PAN data is available.
+func (i *ISO1) Encode(pin string) (string, error) {
+
+	pad, err := i.padding(pin)
+	if err != nil {
+		return "", err
+	}
+
+	// A PIN that is longer than 12 digits is truncated on the right.
+	if len(pin) > 12 {
+		pin = pin[:12]
+	}
```

Comment:
Is there no error or warning to produce?

Reply:
No, it just says take first 12 :D Not sure if there are real cases with pins longer than 6 digits. I'm curious to know.

Reply:
The online docs just say "The VISA-4 PIN block format supports a PIN from 4 to 12 digits in length. A PIN that is longer than 12 digits is truncated on the right." We probably need something specific from the card brands for implementation to get more details.